src/Queue/CachedTransformer.php

```diff
@@ -15,18 +15,37 @@ class CachedTransformer implements QueueItemTransformer
     private $lifetime;
     private $sdk;
     private $serializer;
+    private $shouldCache;
 
-    public function __construct(ApiSdk $sdk, Cache $cache, LoggerInterface $logger, int $lifetime)
-    {
+    public function __construct(
+        ApiSdk $sdk,
+        Cache $cache,
+        LoggerInterface $logger,
+        int $lifetime,
+        callable $shouldCache
+    ) {
         $this->serializer = $sdk->getSerializer();
         $this->sdk = $sdk;
         $this->cache = $cache;
         $this->logger = $logger;
         $this->lifetime = $lifetime;
+        $this->shouldCache = $shouldCache;
+    }
+
+    private function shouldCacheEntity(string $type, string $id)
+    {
+        if ($shouldCache = $this->shouldCache) {
```

Comment:
Ah forgot to make the should cache optional in the constructor, avoids something like `function() { return true; }` in recommendations